app.Makefile

```diff
@@ -49,82 +49,73 @@ app/build:: ;
 app/install:: app/build \
               .build/var/APP_DEPLOYER_IMAGE \
               .build/var/APP_PARAMETERS \
-              .build/var/MARKETPLACE_TOOLS_PATH
+              .build/var/HOME \
+              .build/var/MARKETPLACE_TOOLS_TAG
```

Comment:
This is a good var, thanks!

For local workflows, we need to actually depend on a target that optionally rebuild the dev image.

What do you think about the following, which covers both local workflow and external cases:
- `.build/marketplace/dev` target should be an additional prerequisite here.
- `.build/marketplace/dev` also depends on `MARKETPLACE_TOOLS_TAG`, so it gets rebuilt when the tag changes.
- `MARKETPLACE_TOOLS_TAG` should be computed to something like `local` or `dirty` if it's actually dirty.
- If the tag is `local`/`dirty`, `.build/marketplace/dev` runs `docker build`.
- Otherwise, `.build/marketplace/dev` tries to pull. If it fails to pull, it will build. There are 2 reasons for this: (1) if the commit has been pushed to master, the pull is probably unnecessary (since `docker run` would pull as well) but does no harm; (2) it needs to build unpushed commits.
- There's one caveat here: if the tag goes from `local` to `sha_*` due to the dirty changes being committed, the dev image would be rebuilt while it doesn't need to. That's probably OK.

Note that `app.Makefile` in this case would have to import `marketplace.Makefile`, which it previously didn't need to.

Reply:
For local workflows, I had something similar in mind...

* Partner top level Makefile (e.g. wordpress/Makefile) would append `.build/marketplace/dev` as a dependency to `app/build`.
* `.build/marketplace/dev` runs `docker build` and updates `MARKETPLACE_TOOLS_TAG`.

I didn't include it in this pull request because I wanted to think more about tagging semantics for local development `local` or `dirty` or something. Probably just `local` is sufficient, as I don't think it's necessary to push the dev image anywhere, but wanted to think about it a bit more.

What do you think about this approach?

Reply:
Note: This pattern will break down if a `.build/marketplace/*` image is referenced by a manifest applied to a Kubernetes cluster, as the images are not pushed during local development. This is not a blocker, as these images are not on the install or verify flows.